src/tools/measure.rs

```diff
@@ -0,0 +1,214 @@
+use druid::kurbo::{Circle, Line, ParamCurve, Point, Rect, Size, Vec2};
+use druid::piet::{Color, FontFamily, RenderContext, Text, TextLayout, TextLayoutBuilder};
+use druid::{Data, Env, EventCtx, PaintCtx};
+
+use crate::design_space::DPoint;
+use crate::edit_session::EditSession;
+use crate::mouse::{Drag, Mouse, MouseDelegate, TaggedEvent};
+use crate::tools::{EditType, Tool};
+
+#[derive(Default)]
+pub struct Measure {
+    line: Option<Line>,
+}
+
+const MEASURE_LINE_STROKE_COLOR: Color = Color::rgb8(0x73, 0x9B, 0xCB);
+const MEASURE_INFO_BG_COLOR: Color = Color::rgb8(0x70, 0x70, 0x70);
+const MEASURE_INFO_FG_COLOR: Color = Color::rgb8(0xf8, 0xf8, 0xf8);
+const MEASURE_INFO_ONCURVE_COLOR: Color = Color::rgb8(0x80, 0x80, 0xe0);
+const MEASURE_INFO_OFFCURVE_COLOR: Color = Color::rgb8(0x60, 0xc0, 0x60);
+const MEASURE_INFO_DELTA_COLOR: Color = Color::rgb8(0xa0, 0x20, 0x20);
+const MEASURE_INFO_FONT_SIZE: f64 = 9.0;
+const MEASURE_INTERSECTION_RADIUS: f64 = 3.0;
```

Comment:
A lot of this stuff is styling information that should be shared between widgets, and should ultimately live in the theme or something.